### DISCUSSION TITLE
Instruct NHibernate how NOT to delete base class record

### DISCUSSION THREAD

**Discussion by mwpowellhtx** [nhibernate/nhibernate-core#2565] [General]
I made a design decision in my database to delete a base class table record via after delete trigger on the child class table record. However, what I am finding is that NHibernate first deletes the child record, which is correct, but then also deletes, or attempts to delete, the now-deleted base class record.

At minimum I need to somehow instruct, perhaps via the `Extends<BaseClass>()` call, however, such a mechanism does not presently exist it seems.

I may do something with `SqlDelete()`, provided that would potentially bypass the multiple delete SQL statements being generated, as a workaround.

Worst case I remove the after delete trigger statement(s), but I am loathe to do something like that in this instance because the model integrity is maintained through the database as much as possible.

Open to suggestions, first of all, or potentially does this require some work in either Fluent and/or NHibernate itself?

**Comment by chester89** [comment]
@mwpowellhtx take a look at [this SO question](http://stackoverflow.com/questions/4353126/deleting-a-subclass-deletes-the-parent-as-well). I think you should let ORM do its work. 
Any particular reason you want this done via trigger?

**Comment by mwpowellhtx** [comment]
@chester89 It's not the same question that I'm asking here. Of course, if Child Record gets deleted, Parent Record should be as well. I want for the database to handle its bookkeeping, as I mentioned, and for the ORM, apparently, to be instructed to respect that decision as well.

**Comment by chester89** [comment]
Then I guess it's a question to NHibernate folks. If NH allows this, then
Fluent can support that too.

Sent with MailTrack
https://mailtrack.io/install?source=signature&lang=en&referral=thebitterend77@gmail.com&idSignature=22

2016-09-05 17:33 GMT+03:00 Michael Powell notifications@github.com:

> @chester89 https://github.com/chester89 It's not the same question that
> I'm asking here. Of course, if Child Record gets deleted, Parent Record
> should be as well. I want for the database to handle its bookkeeping, as I
> mentioned, and for the ORM, apparently, to be instructed to respect that
> decision as well.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> https://github.com/jagregory/fluent-nhibernate/issues/357#issuecomment-244760480,
> or mute the thread
> https://github.com/notifications/unsubscribe-auth/AALnfH7LCOpO7aejpBVISqiIwMVJxWzlks5qnCgggaJpZM4J0psP
> .

## 

Yours faithfully,
Gleb

**Comment by mwpowellhtx** [comment]
Thanks, I figured as much. I posted on the NHibernate Dev list (Google) along these lines. For now, I've backed out the triggers.

**Comment by chester89** [comment]
Can you provide a link to the mailing list thread?

Sent with MailTrack
https://mailtrack.io/install?source=signature&lang=en&referral=thebitterend77@gmail.com&idSignature=22

2016-09-05 18:17 GMT+03:00 Michael Powell notifications@github.com:

> Thanks, I figured as much. I posted on the NHibernate Dev list (Google)
> along these lines. For now, I've backed out the triggers.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> https://github.com/jagregory/fluent-nhibernate/issues/357#issuecomment-244769841,
> or mute the thread
> https://github.com/notifications/unsubscribe-auth/AALnfGmuzEo4U4HSlF57w8I5Ijb9URxWks5qnDKmgaJpZM4J0psP
> .

## 

Yours faithfully,
Gleb

**Comment by mwpowellhtx** [comment]
Sure thing. I posted this morning, possibly being "moderated"? I don't know. If/when I see the post, or re-post, will follow up here.

In the meantime, I'm "happy" to report that "it works" when I let the ORM handle things. However, this is as presently expected, as SQL Server Profiler revealed to me earlier.

Thank you.

**Comment by mwpowellhtx** [comment]
[Forever immortalized](http://groups.google.com/forum/#!topic/nhibernate-development/kPKv3Yukj1M). Well, for consideration at any rate.
